tests/helpers/test_collection.py

```diff
@@ -256,6 +256,7 @@ async def test_storage_collection(hass: HomeAssistant) -> None:
 async def test_attach_entity_component_collection(hass: HomeAssistant) -> None:
     """Test attaching collection to entity component."""
     ent_comp = entity_component.EntityComponent(_LOGGER, "test", hass)
+    await ent_comp.async_setup({})
```

Comment:
`async_setup` registers `_async_shutdown` on `EVENT_HOMEASSISTANT_STOP`
Without this, `_async_shutdown` is never called and platforms stay loaded.